File: HighRise/Models/RecipientTable.swift
import Foundation

/// A source-agnostic tabular representation of a recipient list.
///
/// Every importer — CSV, Excel, Word/PDF, Apple Contacts, Outlook — reduces its
/// input to this same shape: a header row plus data rows. That lets one
/// pipeline (`CSVParser.contacts`, email-column detection, the preview table,
/// the merge engine) serve every source without special-casing.
struct RecipientTable: Equatable {
    /// Column headers, in order, original casing preserved.
    let headers: [String]
    /// Data rows; each is padded/truncated to `headers.count` on access.
    let rows: [[String]]

    init(headers: [String], rows: [[String]]) {
        self.headers = headers
        self.rows = rows
    }
}

File: windows/src/HighRise.Core/Models/RecipientTable.cs
namespace HighRise.Core.Models;

/// <summary>
/// A source-agnostic tabular recipient list: a header row plus data rows. Every
/// importer (CSV, Excel, …) reduces its input to this shape so one pipeline can
/// serve every source. Port of the Swift <c>RecipientTable</c>.
/// </summary>
public sealed class RecipientTable
{
    /// <summary>Column headers, in order, original casing preserved.</summary>
    public IReadOnlyList<string> Headers { get; }

    /// <summary>Data rows; a row may be shorter than <see cref="Headers"/>.</summary>
    public IReadOnlyList<IReadOnlyList<string>> Rows { get; }

    public RecipientTable(IReadOnlyList<string> headers, IReadOnlyList<IReadOnlyList<string>> rows)
    {
        Headers = headers;
        Rows = rows;
    }
}
